framework/global/timer.h

```diff
@@ -0,0 +1,112 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2020 MuseScore BVBA and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2.
+//
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+//  GNU General Public License for more details.
+//
+//  You should have received a copy of the GNU General Public License
+//  along with this program; if not, write to the Free Software
+//  Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+//=============================================================================
+#ifndef MU_FRAMEWORK_TIMER_H
+#define MU_FRAMEWORK_TIMER_H
+
+#include "async/notification.h"
+#include "async/asyncable.h"
+#include <chrono>
+#include <memory>
+#include <thread>
+
+namespace mu {
+/*!
+ * mu::Timer
+ * usage:
+ *      auto timer = new Timer(std::chrono::microseconds(20));
+ *      timer.onTimeout(this, []() { LOGI() << "Timer call";});
+ *      timer.run();
+ */
+class Timer
+{
+public:
+    using time_t = std::chrono::microseconds;
+
+    explicit Timer(time_t interval)
+        : m_interval(interval), m_active(false) {}
+
+    ~Timer() { stop(); }
+
+    //! start timer's thread
+    void run()
+    {
+        if (!isRunning()) {
+            m_active = true;
+            m_thread.reset(std::move(new std::thread([this]() {
+                    timerLoop();
+                })));
+            m_thread->detach();
+            m_started = std::chrono::steady_clock::now();
+        }
+    }
+
+    //! stop and kill the timer's thread
+    void stop()
+    {
+        m_active = false;
+        if (m_thread) {
+            m_thread.release();
+        }
+    }
+
+    //! return current timer's status
+    bool isRunning() const
+    {
+        return m_active;
+    }
+
+    //! add notification on timer's interval
+    template<typename Func>
+    void onTimeout(const async::Asyncable* receiver, Func function)
+    {
+        m_notification.onNotify(receiver, function);
+    }
+
+    //! return microseconds since last run
+    float secondsSinceRun()
```

Comment:
Can you explain the purpose of this method?

Reply:
There was a little mess in the naming. I've cleaned it up.
Thanks